src/SerilogTracing.Instrumentation.AspNetCore/Instrumentation/AspNetCore/HttpRequestInActivityInstrumentor.cs

```diff
@@ -152,13 +152,37 @@ public void InstrumentActivity(Activity activity, string eventName, object event
                 // 2. Clients that don't send any traceparent header may still produce a recorded activity
                 CreateReplacementActivity(incoming, true, true, true, true),
             
-            _ => throw new ArgumentOutOfRangeException()
+            _ => throw new ArgumentOutOfRangeException(nameof(incomingTraceParent))
         };
     }
 
     static Activity? CreateReplacementActivity(Activity? incoming, bool inheritTags, bool inheritParent, bool inheritFlags, bool inheritBaggage)
     {
-        var replacement = ReplacementActivitySource.CreateActivity(DefaultActivityName, ActivityKind.Server);
+        // The `incoming` activity is the one ASP.NET Core generated by default. We're only interested in its parent if
+        // there is one. Switching off `inheritParent` when there isn't, prevents us from trying to override a nonexistent
+        // sampling decision a little further down. Checking `HasRemoteParent` would be useful here, but it creates
+        // problems for unit testing.
+        inheritParent = inheritParent && incoming != null &&
+                        incoming.ParentSpanId.ToHexString() != default(ActivitySpanId).ToHexString();
+
+        // Unfortunately there's no way to access the parent's trace flag apart from parsing the header ourselves,
+        // which we'll eventually need to implement. This gets us there for now.
+        var flags = ActivityTraceFlags.None;
+        if (inheritParent && inheritFlags)
+        {
+            if (incoming?.ParentId?.EndsWith("-01") == true)
```

Comment:
The format isn't that complicated, would it be worth at least pulling this into a utility class so we can track where we'd need to do it properly over time?

Reply:
👍